kernel/thread.c

```diff
@@ -903,3 +904,14 @@ static inline void z_vrfy_k_thread_abort(k_tid_t thread)
 #include <syscalls/k_thread_abort_mrsh.c>
 
 #endif /* CONFIG_USERSPACE */
+
+#ifdef CONFIG_IRQ_OFFLOAD
+static K_SEM_DEFINE(offload_sem, 1, 1);
+
+void irq_offload(irq_offload_routine_t routine, void *parameter)
+{
+	k_sem_take(&offload_sem, K_FOREVER);
```

Comment:
Question: Would it make sense to only use the semaphore if SMP? I guess otherwise it is just overhead?

Reply:
this is a (much less likely) concern for UP as well.
irq_offload() is for test cases, it only exists to validate kernel objects in IRQ context.